src/manifest.c

```diff
@@ -679,38 +684,54 @@ struct manifest *load_mom_err(int version, bool latest, bool mix_exists, int *er
  * Note that if the manifest fails to download, or if the manifest fails to be
  * loaded into memory, this function will return NULL.
  */
-struct manifest *load_manifest(int current, int version, struct file *file, struct manifest *mom, bool header_only)
+struct manifest *load_manifest(int current, int version, struct file *file, struct manifest *mom, bool header_only, bool delta_manifest)
 {
 	struct manifest *manifest = NULL;
+	int delta_manifest_version = 0;
 	int ret = 0;
 	bool retried = false;
 
 retry_load:
-	ret = retrieve_manifests(current, version, file->filename, file, file->is_mix);
-	if (ret != 0) {
-		fprintf(stderr, "Failed to retrieve %d %s manifest\n", version, file->filename);
-		return NULL;
-	}
+	if (delta_manifest) {
+		if (file->peer != NULL) {
+			delta_manifest_version = file->peer->last_change;
```

Comment:
I am going to condense this to one condition statement:

if (delta_manifest && file->peer != NULL), use delta manifest
else, don't use delta manifest

I can also check for the case when the manifest is unchanged and skip attempting to load a delta manifest.